rust/agama-network/src/dbus/interfaces.rs

```diff
@@ -0,0 +1,402 @@
+//! Network D-Bus interfaces.
+//!
+//! This module contains the set of D-Bus interfaces that are exposed by [D-Bus network
+//! service](crate::NetworkService).
+use super::ObjectsRegistry;
+use crate::{
+    action::Action,
+    error::NetworkStateError,
+    model::{Connection as NetworkConnection, Device as NetworkDevice, WirelessConnection},
+};
+use parking_lot::{MappedMutexGuard, Mutex, MutexGuard};
+use std::{
+    net::{AddrParseError, Ipv4Addr},
+    sync::{mpsc::Sender, Arc},
+};
+use uuid::Uuid;
+use zbus::{dbus_interface, zvariant::ObjectPath};
+
+/// D-Bus interface for the network devices collection
+///
+/// It offers an API to query the devices collection.
+pub struct Devices {
+    objects: Arc<Mutex<ObjectsRegistry>>,
+}
+
+impl Devices {
+    /// Creates a Devices interface object.
+    ///
+    /// * `objects`: Objects paths registry.
+    pub fn new(objects: Arc<Mutex<ObjectsRegistry>>) -> Self {
+        Self { objects }
+    }
+}
+
+#[dbus_interface(name = "org.opensuse.Agama.Network1.Devices")]
+impl Devices {
+    /// Returns the D-Bus paths of the network devices.
+    pub fn get_devices(&self) -> Vec<ObjectPath> {
+        let objects = self.objects.lock();
+        objects
+            .devices_paths()
+            .iter()
+            .filter_map(|c| ObjectPath::try_from(c.clone()).ok())
+            .collect()
+    }
+}
+
+/// D-Bus interface for a network device
+///
+/// It offers an API to query basic networking devices information (e.g., the name).
+pub struct Device {
+    device: NetworkDevice,
+}
+
+impl Device {
+    /// Creates an interface object.
+    ///
+    /// * `device`: network device.
+    pub fn new(device: NetworkDevice) -> Self {
+        Self { device }
+    }
+}
+
+#[dbus_interface(name = "org.opensuse.Agama.Network1.Device")]
+impl Device {
+    #[dbus_interface(property)]
+    pub fn name(&self) -> &str {
+        &self.device.name
+    }
+
+    #[dbus_interface(property)]
+    pub fn device_type(&self) -> u8 {
```

Comment:
Why not simple type? Device is already in interface name

Reply:
Because `type` is a reserved word in Rust. I would rename the D-Bus name but I would keep the if the function (unless we find a better name for all *types*).

Reply:
fine for me. I am talking here mainly about dbus API